internal/provider/fwprovider/auth.go

```diff
@@ -0,0 +1,61 @@
+package fwprovider
```

Comment:
nit: sorry I forgot to add the copywrite headers for these! Definitely can let the copywrite bot take care of these before we merge to main, but just wanted to add a comment here to keep track. not a blocker